pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java

```diff
@@ -766,11 +772,25 @@ public static void testQuery(@Nonnull String pinotQuery, @Nonnull String queryFo
           }
           String actualValue = actualValueBuilder.toString();
 
-          // Check actual value in expected values set
-          if (!expectedValues.contains(actualValue)) {
-            String failureMessage = "Selection result returned in Pinot but not in H2: " + actualValue;
-            failure(pinotQuery, sqlQueries, failureMessage);
-            return;
+          final BrokerRequest brokerRequest = PinotQueryParserFactory.get("pql").compileToBrokerRequest(pinotQuery);
+          final List<String> selectionColumns = brokerRequest.getSelections().getSelectionColumns();
+          final Set<String> orderByColumns = getOrderByColumns(brokerRequest.getOrderBy());
+
+          final List<SelectionSort> orderBy = brokerRequest.getOrderBy();
+          if ((orderBy != null) && (!orderBy.isEmpty()) && selectionColumns.containsAll(orderByColumns))  {
+            // Check actual value in expected values set
+            if (!expectedValues.get(rowIndex).equals(actualValue)) {
```

Comment:
We need to handle the tie case

Reply:
it's impossible from the results. E.g. `select a from table, order by b,c`, if the tie results more than limit size, then we don't event know the ordering.

The only thing we can determine this is that order by columns are the same as selection columns.